auth/auth-router.js

```diff
@@ -0,0 +1,72 @@
+require("dotenv").config();
+const bcrypt = require("bcryptjs");
+const express = require("express");
+const jwt = require("jsonwebtoken");
+
+const router = require("express").Router();
+const db = require("../users/users-model");
+
+router.post("/register", (req, res) => {
+  const newUser = req.body;
+
+  if (!newUser.password || !newUser.username) {
+    res.status(400).json({
+      message: "Must provide username and password",
+    });
+  } else {
+    const hash = bcrypt.hashSync(newUser.password, 15);
+    newUser.password = hash;
+
+    db.add(newUser)
+      .then((user) => {
+        res.status(201).json(user);
+      })
+      .catch((error) => {
+        res.status(500).json({
+          errorMessage: "Error with server",
+        });
+      });
+  }
+});
+
+router.post("/login", (req, res) => {
+  const { username, password } = req.body;
+
+  db.findBy({ username })
+    .then((user) => {
+      if (user && bcrypt.compareSync(password, user[0].password)) {
+        const token = generateToken(user);
```

Comment:
`user` appears to be an array based on how it is used with `user[0]` throughout the rest of this function however when passed into `generateToken()` it gets used as an object with reference to `user.id` and `user.username`

In one of these scenarios something is going to be undefined.
Can you confirm if `user` is an array and if so pass in `user[0]` here?
Otherwise if it is not an array remove the `[0]` in the other locations?